### ISSUE TITLE
Support Multiple Languages

### ISSUE THREAD

**Issue by hql287** [hql287/Manta#9]
**UPDATE:**

Thank you very much for your patience, now you can help translate Manta on [Crowdin](https://crowdin.com). 

[Please follow this instruction to start translating the app](https://github.com/hql287/Manta/wiki/Translating-Manta).

Thanks!

**Original Post:**

I think this would be a very nice feature to have. 
But first, let me know which languages you would like Manta to support.

[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/France)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/France/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Spanish)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Spanish/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Portuguese)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Portuguese/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Chinese)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Chinese/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Japanese)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Japanese/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Korean)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Korean/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Italian)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Italian/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Russian)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Russian/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Vietnamese)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Vietnamese/vote)
[![](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Thai)](https://api.gh-polls.com/poll/01C10PCX59FS1ZNX9T813Q2PSM/Thai/vote)
[![](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/German)](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/German/vote)
[![](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Dutch)](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Dutch/vote)
[![](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Brazilian%20Portuguese)](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Brazilian%20Portuguese/vote)
[![](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Turkish)](https://api.gh-polls.com/poll/01C32ZRKE5KJ746Y6HN65VGRN5/Turkish/vote)

Did I miss something? Pls let me know in the comment.

**Comment by SkyzohKey** [comment]
Would be nice to have, especially being able to set the texts on the invoice. I'm french and can translate the whole in French if you provides a way to :p

**Comment by ghazal** [comment]
+1 for this.

**Comment by jmrodriguesgoncalves** [comment]
I could give this a shot as well. I speak Portuguese, so I could do translation for that language if you want?

**Comment by hql287** [comment]
@jmrodriguesgoncalves Sure, that would be really great! But I need to find a better way to implement this feature first. It actually was planned for the first release but I was not really happy with the solution, (some feel too hacky, some too overkill) so decided to skip it.

**Comment by dertuerke** [comment]
Germany u forgot ^^

**Comment by SkyzohKey** [comment]
You can try that awesome React lib by @Yahoo => https://github.com/yahoo/react-intl

**Comment by hql287** [comment]
@SkyzohKey: I did and it seemed overkill to me. But probably I'll still give it another go! Thanks!

**Comment by erhankaradeniz** [comment]
I'd like support for Dutch

**Comment by OrionRandD** [comment]
We are in great need for having Brazilian Portuguese added to it as there are lots of small and medium start-ups using Open-source software over here...

**Comment by OrionRandD** [comment]
Also it would be good if there were a field to fill in the Receiver bank account number
Thx for the great app.

**Comment by JJYing** [comment]
I would love to help translating Manta to Chinese!

**Comment by OrionRandD** [comment]
@hql287 perhaps aspell or enchant can help...
Also it would be good if there were a field to fill in the Receiver bank account number
Thx for the great app.

**Comment by hql287** [comment]
@OrionRandD: Can you create a separate issue on this and explain what it does and how it should be implemented? Thanks!

**Comment by davidsommer** [comment]
German, i would provide the first draft of translation

**Comment by SjamonDaal** [comment]
Dutch please, I already offered to translate :+1:

**Comment by hql287** [comment]
Just added some new options:

* German @davidsommer, @dertuerke 
* Dutch @SjamonDaal @erhankaradeniz 
* Brazilian Portuguese @OrionRandD  

I had to create a new poll since GitHub Polls doens't let me edit the current one. The percentage might not be accurate so just ignore it.

**Comment by c0b41** [comment]
Turkish 👊

**Comment by hql287** [comment]
![Yes](https://media.giphy.com/media/S3Ot3hZ5bcy8o/giphy.gif)
@c0b41 Sure!

**Comment by lajlev** [comment]
Danish. I will help you do the translations 🤗

**Comment by henriquemacedo** [comment]
And I can help you with portuguese (pt-PT) translation 👊

**Comment by DarkSmile92** [comment]
I'll do the german translation 👍

**Comment by hql287** [comment]
Thank you all very much for your patience. I've updated the issue with detail instruction on how to translate Manta on [Crowdin](https://crowdin.com/profile)

Please let me know if you have any issue.
Thanks! 🤗

**Comment by henriquemacedo** [comment]
@hql287 🇵🇹 done!

**Comment by DarkSmile92** [comment]
German done 👍

**Comment by SjamonDaal** [comment]
Dutch done! 🙂

**Comment by hql287** [comment]
@SjamonDaal: I think each translation should have at least 1 upvote in order to be approved. Can you get another Dutch speaking dev to review the translation? Thanks!

**Comment by eimantaaas** [comment]
Lithuanian would be great. I can help with this translation.

**Comment by SjamonDaal** [comment]
@hql287 Sure, I'll ask my colleague to translate the file! Will be done tonight. 😉

**Comment by hql287** [comment]
@SjamonDaal Awesome, thanks! 
@eimantaaas: That's great. I just [added Lithuanian](https://crowdin.com/project/manta/lt) per your request. Thanks! 👍

**Comment by nlaubres** [comment]
🇫🇷 completed !

**Comment by hql287** [comment]
Hate to break it to you but I just added 2 more files 😅 (and there'are still a few more to come).

**Comment by SjamonDaal** [comment]
@hql287
Well okay then 😋 I'll continue!

Common.json
Item 5, Phone Numner
Small typo

**Comment by nlaubres** [comment]
Small typo >
settings.json
Item 17 "Open PDF after Exprt" instead of [..]Export

**Comment by henriquemacedo** [comment]
@joaotmdias can you check ([here](https://crowdin.com/project/manta/pt-PT#)) the portuguese translation?

**Comment by filipkominik** [comment]
Can you add Czech language support? 🇨🇿

**Comment by hql287** [comment]
Yes, will you be able to translate?

**Comment by rcorzogutierrez** [comment]
:es: completed!

**Comment by hql287** [comment]
Just added `invoices` translation :)

**Comment by codedge** [comment]
🇩🇪 completed!

**Comment by filipkominik** [comment]
@hql287 yes - I will try to translate it ;)

**Comment by hql287** [comment]
Thanks, @codedge 👍

Also, I just added Czech. You can [start translating here](https://crowdin.com/project/manta/cs# ). Thanks!

**Comment by filipkominik** [comment]
@hql287 🇨🇿 completed - but will appreciate any corrections from others (not sure if everything is translated correctly)

**Comment by hql287** [comment]
@filipkominik Awesome! Can you ask any friend of yours who can speak Czech to check? It's very easy to join, please [follow this guide](https://github.com/hql287/Manta/wiki/Translating-Manta). 

Let me know if this works. 
Thanks

**Comment by JJYing** [comment]
Chinese 100% completed

**Comment by hql287** [comment]
Thanks, Ying 👍.  Is it possible for you to get another Chinese dev to check the translation?

**Comment by JJYing** [comment]
@hql287 Let me spread the word to see if there is anyone to help   :D

**Comment by JJYing** [comment]
By the way, I found that the strings appears in the Invoice templates are not in the translation system and they are directly written in the jsx template files, is it possible to also localize the invoice themselves by selecting language options?

**Comment by hql287** [comment]
@JJYing : I'm working on it at the moment 😄. You can track the progress with [these little checkboxes](https://github.com/hql287/Manta/pull/187#issue-290540828). 

Btw, I just checked out your portfolio, very impressive!!! 👍

**Comment by hql287** [comment]
@SjamonDaal Sorry, I just uploaded the last 2 files 😜. 
Thanks again for your help 👍

**Comment by hql287** [comment]
Alright, 6 languages completed so far. That's awesome guys! 😘

- [ ] 🇩🇪 German 
- [ ] 🇳🇱 Dutch
- [ ] 🇫🇷 French
- [ ] 🇪🇸 Spanish
- [ ] 🇨🇳 Chinese Simplified
- [ ] 🇻🇳 Vietnamese

**Comment by bjo3rns** [comment]
@hql287 when do you integrate the other languages?

**Comment by hql287** [comment]
@bjo3rns I'll integrate all the languages once they have the approved translations. Would you like to help? [Please sign up as a translator here](https://translate.manta.life).

**Comment by hql287** [comment]
Guys, I just made some of you proofreaders on Crowdin.

@SjamonDaal:  Can you approve the Dutch translation?
@JJYing: Can you approve the Chinese translation? 
@henriquemacedo: Can you approve the Portuguese translation?
@filipkominik: Can you approve the Czech translation?
@rcorzogutierrez: Can you approve the Spanish translation?
@DarkSmile92: Can you approve German translation?
@nlaubres: Can you approve French translation? 

(... I think I should've made a loop for this comment 😜)

Anyway, still need more people to approve Japanese, Korean and others. Please let me know if you're interested in being a proofreader and your username on Crowdin.

Thanks!!!

**Comment by hql287** [comment]
Just added the language names for translation, (sorry, I forgot) this should be the last thing to translate before we can add a new language to Manta.

**Comment by henriquemacedo** [comment]
@joaotmdias can you check the portuguese translation?

**Comment by hql287** [comment]
@davidsommer: I noticed German is completed, can you help me to approve the translation (just made you the proofreader)? Thanks!

**Comment by isoppp** [comment]
@hql287 I can try Japanese if you need 😄

**Comment by hql287** [comment]
@isoppp That would be great, please do! :)

**Comment by isoppp** [comment]
Japanese done. But need more check and vote X<

**Comment by hql287** [comment]
@isoppp Just made you the proofreader for Japanese. Would you be able to get someone else also speaks Japanese to double check your translation before approving them?

**Comment by isoppp** [comment]
@hql287
Sure thing.
I try ask friends check translations :D

**Comment by marcelblanarik** [comment]
Hey, I could translate Manta to Slovak language :)

**Comment by hql287** [comment]
@marcelblanarik Hey Marcel, thanks for being interested in translating Manta. Have you signed up on Crowdin yet? If yes, please let me know your username so I can set you up as the proofreader for Slovak. Thanks! 👍

**Comment by marcelblanarik** [comment]
@hql287 Hey Hung, my username on Crowdin is marcel251. Thanks

**Comment by hql287** [comment]
@marcelblanarik Please join Manta on Crowdin first so I can set you up as the proofreader. Thanks!

**Comment by marcelblanarik** [comment]
@hql287 Done.

**Comment by marcelblanarik** [comment]
@hql287 Slovak translation is done. 👍

**Comment by hql287** [comment]
@marcelblanarik Yes, just noticed it. You're fast, bro!!! 👍

**Comment by marcelblanarik** [comment]
I'm glad that I could help. And OK, I'll ask my friend to check it.

**Comment by marcelblanarik** [comment]
@hql287 Hey, I asked my friend to check Slovak translation, he made a few changes, I approved them. Please check it. Thanks

**Comment by hql287** [comment]
@marcelblanarik That's great! But I see only 8% of the translation are approved?

**Comment by marcelblanarik** [comment]
@hql287 Ah, sorry, I'm new to Crowdin, now it should be alright. :)

**Comment by hql287** [comment]
@marcelblanarik Nice work, bro! 👍 Thanks!

**Comment by marcelblanarik** [comment]
No problem 👍

**Comment by rrussi** [comment]
Hey, Italian translation is done but not approved...

**Comment by hql287** [comment]
@rrussi Yes, I just noticed that as well. @stefanomarra is current the proofreader for Italian at the moment, you guys might want to have a little chat with each other :)

**Comment by stefanomarra** [comment]
Hi, @rrussi, I will go through all the texts today and approve the ones that are good to go. There are however some texts that maybe could be improved or upvoted. If you could help it would be really great.

**Comment by rrussi** [comment]
Ok @stefanomarra, if you approve texts you think are good then we can chat about the ones that can be improved.

**Comment by aloukissas** [comment]
Can help translate in Greek (native speaker, bi-lingual Gr/En).

**Comment by hql287** [comment]
Hey Alex, thanks for being interested in translating Manta. Greek has already been added to the list and you can start translating it here: https://crowdin.com/project/manta/el

Let me know if you have any question. 

Thanks!

**Comment by rrussi** [comment]
Hi, @stefanomarra. The work on the italian translation has stopped... How could I help to go ahead and close the translation?

**Comment by stefanomarra** [comment]
Hi @rrussi, sorry for the delay, I will close the italian translation in the next days. I have to check some texts in the app to see if they fit right or not, before approving them. I will follow up here once I'll approve them and we are ready to go.

**Comment by stefusilviu** [comment]
Hello @hql287 . Thanks for this great app. I found it trying to do this by myself but you did a great job. Please set up the Romanian language. I can translate it. Thank you again!

**Comment by hql287** [comment]
@stefanomarra: Thanks for the kind word! Just added Romanian per your request. You can start translating here: https://crowdin.com/project/manta/ro. 

Let me know if you have any question.

**Comment by stefanomarra** [comment]
Above message from @hql287 is for you @stefusilviu ;)
Regarding the italian translation it's ready to be added to the next release @rrussi

**Comment by hql287** [comment]
@stefanomarra: Ahh, sorry bro! 😜

**Comment by stefusilviu** [comment]
@hql287  It seems that I don't have such an option.

**Comment by hql287** [comment]
@stefusilviu: Sorry, what I meant was aking if you would be up for it :) Anyway, I just made you the proofreader for Romanian. Thanks!

**Comment by hql287** [comment]
@stefusilviu: Thank YOU!

**Comment by Cuchulain** [comment]
Czech

**Comment by rlisowski** [comment]
@hql287 can we have Polish translation on crowdin.com? I can help with this one.

**Comment by ForsbergOne** [comment]
@hql287  I can help translate the app to Swedish.

**Comment by ppsirius** [comment]
Polish, I try to help

**Comment by ItamarBittton** [comment]
@hql287  I can help translate the app to Hebrew

**Comment by P1X3L0V4** [comment]
@hql287 I'd like to help with Polish translation

**Comment by maltitco** [comment]
@hql287 Please add Polish language to crowdin.
Already several people ask about it and I should not have a problem with the translation.

**Comment by CosminBd** [comment]
@hql287  the Romanian translation 100% done as far as i see on Crowdin, can you make it available?

**Comment by adifaidz** [comment]
@hql287 Please add Malay language to crowdin, I can contribute to it

**Comment by abeeribrahim** [comment]
Please add Arabic, I'll be very happy to help in translation...

**Comment by robgha01** [comment]
@hql287 Please add Swedish so i can translate :)

**Comment by ksadjad** [comment]
@hql287 Please add Persian so I can translate it to

**Comment by hnhtag** [comment]
I'd like to help with Myanmar (Burmese) translation.

**Comment by ahmedfouzan** [comment]
@hql287 I could help with Urdu.

**Comment by victor141516** [comment]
Could it be possible, for us to test translations, to download the translated files from Crowdin and add them to the app somehow?

**Comment by qpixHassaballah** [comment]
Arabic

**Comment by tomasbal** [comment]
Macedonian, I already did PR.

**Comment by stefusilviu** [comment]
@hql287 Any news on adding the new translations that are done already?  For example romanian language.

**Comment by trisnakurniawan** [comment]
add indonesia please

**Comment by beardoweirdo** [comment]
Can we get Hindi as well?

Thanks.

**Comment by danvaly** [comment]
Hi, I speak Romanian I would love to help you on this. What do you think

**Comment by christiankarlsson** [comment]
Swedish
